### ISSUE TITLE
Add a mobile-friendly view of the public scoreboard

### ISSUE THREAD

**Issue by meisterT** [DOMjudge/domjudge#2401]
The public scoreboard is probably mostly viewed on mobile devices, so a mobile-friendly view with reduced information would make sense.

I did some prototyping to check feasibility in 
https://github.com/DOMjudge/domjudge/commit/fbe6199364a55b413ab0eb1b84692affeea28c82

It would roughly look like this:
![image](https://github.com/DOMjudge/domjudge/assets/418721/ff23155e-1ee1-4290-88f5-b37894fc8457)

As the code is obviously in no shape to being merged, I am interested in feedback whether this is the right direction, whether we need to add more info, remove some more info or present it differently.

**Comment by iMylene** [comment]
Is it possible to remove the logos of institutions? Does that make the scoreboard compacter? I do miss the favorite button in this setup, maybe the button can go on the logo spot?

**Comment by nickygerritsen** [comment]
I heard from people that logo's add a touch of niceness, so not sure if that is the best move?

**Comment by meisterT** [comment]
> Is it possible to remove the logos of institutions? Does that make the scoreboard compacter? I do miss the favorite button in this setup, maybe the button can go on the logo spot?

I think we could bring back the :heart: in some form

**Comment by nickygerritsen** [comment]
I did some changes locally, but can't push to your branch since there is no PR (and thus it's not allowed).

Basically I did:

- Make team names not fall of the screen by calculating their max width.
- Make non mobile scoreboard not left-aligned.
- Add a bit of margin to the header.
- Make the problem boxes right aligned.
- Change the card at the top to show 2 lines (name + contestt time) on mobile.

Screenshots:

iPhone 15 Pro Max:
<img width="512" alt="Screenshot 2024-04-03 at 16 10 15" src="https://github.com/DOMjudge/domjudge/assets/550145/f1acb484-cc14-4d98-8ffe-32fa23111fe8">

iPhone 15 Pro:
<img width="559" alt="Screenshot 2024-04-03 at 16 10 19" src="https://github.com/DOMjudge/domjudge/assets/550145/0bb9d3cd-57b1-4e5a-9713-78f3c34bd074">

Note that if you have a screen with less width than the iPhone 15, the problem boxes might not fit. Not sure if we want to handle that case.
